engine/src/main/java/org/terasology/engine/rendering/world/RenderableWorldImpl.java

```diff
@@ -386,9 +372,44 @@ public int queueVisibleChunks(boolean isFirstRenderingStageForCurrentFrame) {
 
         if (isFirstRenderingStageForCurrentFrame) {
             for (Chunk chunk : chunksInProximityOfCamera) {
-                if (isChunkValidForRender(chunk) && (chunk.isDirty() || !chunk.hasMesh())) {
+                if (isChunkValidForRender(chunk) && (chunk.isDirty() || !chunk.hasMesh()) && !chunkMeshProcessing.containsKey(chunk.getPosition())) {
                     statDirtyChunks++;
-                    chunkMeshUpdateManager.queueChunkUpdate(chunk);
+                    final Chunk currentChunk = chunk;
+                    chunkMeshProcessing.put(chunk.getPosition(), Observable
+                            .just(currentChunk)
+                            .observeOn(Schedulers.io())
+                            .delay(500 * processedChunks, TimeUnit.MILLISECONDS) // delay chunks so everything is not all evaluated at once
```

Comment:
stagger the evaluation so all the chunks are not all submitted at once. helps with keeping frametimes up.

Reply:
How big is `processedChunks` on average? Delaying by half a second multiplied with that number could be quite some time...